src/components/ContentSidebar/ActivityFeed/activity-feed/ActivityFeed.js

```diff
@@ -424,30 +424,45 @@ class ActivityFeed extends React.Component<Props, State> {
     }
 
     /**
-     * Adds a versions entry if the current file version was restored from a previous version
+     * Adds the current version from the file object, which may be a restore
      *
      * @param {FileVersions} versions - API returned file versions for this file
-     * @return {FileVersions} modified versions array including the version restore
+     * @return {FileVersions} modified versions array including the current/restored version
      */
-    addRestoredVersion(versions: FileVersions) {
-        const { file } = this.props;
+    addCurrentVersion(versions: FileVersions, file: BoxItem) {
         const { restored_from, modified_at, file_version } = file;
 
-        // Ensures restored version is only added on first feed loads
-        const lastVersion = versions.total_count ? versions.entries[versions.total_count - 1] : {};
-        if (restored_from && lastVersion.action !== VERSION_RESTORE_ACTION) {
-            const restoredVersion = versions.entries.find((version) => version.id === restored_from.id);
-
-            if (restoredVersion) {
-                versions.entries.push({
-                    ...restoredVersion,
-                    // $FlowFixMe
-                    id: file_version.id,
-                    created_at: modified_at,
-                    action: VERSION_RESTORE_ACTION
-                });
-                versions.total_count += 1;
+        if (file_version) {
+            const { modified_by, version_number } = file;
```

Comment:
if you're pulling the remaining props from file here, maybe only get `file_version` at the top of the method and then get `restored_from`, `modified_at`, `version_number`, etc here?